model.py

```diff
@@ -3396,69 +3396,47 @@ def all_descriptions(self):
                 Resource.quality.desc())
 
 
-    def set_primary_edition(self):
-        """Which of this Work's Editions should be used as the default?
-        """
+    def set_primary_edition(self, new_primary_edition):
+        self.primary_edition = new_primary_edition
 
-        # for each edition, see if its LicensePool was superceded or suppressed
-        # if yes, it's not the champion
-        for edition in self.editions:
-            # There can be only one.
-            if edition != self.primary_edition:
-                edition.is_primary_for_work = False
-            else:
-                edition.is_primary_for_work = True
-                # let the edition know it's attached to this work now
-                edition.work = self
-                # let the edition's pool know they have a work
-                if edition.is_presentation_for:
-                    edition.is_presentation_for.work = self
 
-        
-        WorkCoverageRecord.add_for(
-            self, operation=WorkCoverageRecord.CHOOSE_EDITION_OPERATION
-        )
 
-
-    def calculate_presentation(self, policy=None, search_index_client=None):
-        """Make a Work ready to show to patrons.
+    def calculate_primary_edition(self, policy=None, pool_to_not_check=None):
+        """ Which of this Work's Editions should be used as the default?
```

Comment:
pool_to_not_check doesn't seem to be used and I'm not sure why you would exempt one and only one license pool from consideration.